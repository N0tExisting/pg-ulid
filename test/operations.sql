BEGIN;

CREATE EXTENSION ulid;

SELECT ('7ZZZZZZZZZZZZZZZZZZZZZZZZZ'::ulid)::timestamp;
SELECT ('00000000000000000000000000'::ulid)::timestamp;
SELECT ('01H55TNAQ96WPSWE6WZRCH9G0C'::ulid)::timestamp;

SELECT '01H539HMN700009Y99Q81VWK04'::ulid = '01H539HMN700009Y99Q81VWK04'::ulid;
SELECT '01H539HMN700009Y99Q81VWK04'::ulid >= '01H539HMN700009Y99Q81VWK04'::ulid;
SELECT '01H539HMN700009Y99Q81VWK04'::ulid > '01H539HMN700009Y99Q81VWK04'::ulid;
SELECT '01H539HMN700009Y99Q81VWK04'::ulid <= '01H539HMN700009Y99Q81VWK04'::ulid;
SELECT '01H539HMN700009Y99Q81VWK04'::ulid < '01H539HMN700009Y99Q81VWK04'::ulid;

CREATE TABLE ulids (id ulid);

INSERT INTO ulids
VALUES
  ('7ZZZZZZZZZZZZZZZZZZZZZZZZZ'),
  ('0B7RR1YKW8M15S17RGPJ287K4T'),
  ('00000000000000000000000000'),
  ('3GDQGKJ6TH887R4YHZ4KGABYF3'),
  ('01H58HW7T70008RS65ZCAH32KF'),
  ('013Z8SSES7Z1YEXRSDPP8GC4R7');

SELECT id, id::timestamp as ts FROM ulids ORDER BY id;

ROLLBACK;
