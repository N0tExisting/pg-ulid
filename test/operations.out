BEGIN;
BEGIN
CREATE EXTENSION ulid;
CREATE EXTENSION
SELECT ('7ZZZZZZZZZZZZZZZZZZZZZZZZZ'::ulid)::timestamp;
        timestamp         
--------------------------
 10889-08-02 05:31:50.655
(1 row)

SELECT ('00000000000000000000000000'::ulid)::timestamp;
      timestamp      
---------------------
 1970-01-01 00:00:00
(1 row)

SELECT ('01H55TNAQ96WPSWE6WZRCH9G0C'::ulid)::timestamp;
        timestamp        
-------------------------
 2023-07-12 19:53:43.401
(1 row)

SELECT '01H539HMN700009Y99Q81VWK04'::ulid > '01H539HMN700009Y99Q81VWK04'::ulid;
 ?column? 
----------
 f
(1 row)

SELECT '01H539HMN700009Y99Q81VWK04'::ulid >= '01H539HMN700009Y99Q81VWK04'::ulid;
 ?column? 
----------
 t
(1 row)

SELECT '01H539HMN700009Y99Q81VWK04'::ulid > '01H539HMN700009Y99Q81VWK04'::ulid;
 ?column? 
----------
 f
(1 row)

SELECT '01H539HMN700009Y99Q81VWK04'::ulid <= '01H539HMN700009Y99Q81VWK04'::ulid;
 ?column? 
----------
 t
(1 row)

SELECT '01H539HMN700009Y99Q81VWK04'::ulid < '01H539HMN700009Y99Q81VWK04'::ulid;
 ?column? 
----------
 f
(1 row)

CREATE TABLE ulids (id ulid);
CREATE TABLE
INSERT INTO ulids
VALUES
  ('7ZZZZZZZZZZZZZZZZZZZZZZZZZ'),
  ('0B7RR1YKW8M15S17RGPJ287K4T'),
  ('00000000000000000000000000'),
  ('3GDQGKJ6TH887R4YHZ4KGABYF3'),
  ('01H58HW7T70008RS65ZCAH32KF'),
  ('013Z8SSES7Z1YEXRSDPP8GC4R7');
INSERT 0 6
SELECT id, id::timestamp as ts FROM ulids ORDER BY id;
             id             |            ts            
----------------------------+--------------------------
 00000000000000000000000000 | 1970-01-01 00:00:00
 013Z8SSES7Z1YEXRSDPP8GC4R7 | 2009-03-04 13:12:03.623
 01H58HW7T70008RS65ZCAH32KF | 2023-07-13 21:17:55.911
 0B7RR1YKW8M15S17RGPJ287K4T | 2361-09-23 12:00:36.232
 3GDQGKJ6TH887R4YHZ4KGABYF3 | 5887-04-11 00:06:02.321
 7ZZZZZZZZZZZZZZZZZZZZZZZZZ | 10889-08-02 05:31:50.655
(6 rows)

ROLLBACK;
ROLLBACK
